gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java

```diff
@@ -147,9 +146,10 @@ public static Schema setSchemaCreationTime(Schema inputSchema, String creationTi
    * Common use cases for this method is in traversing {@link Schema} object into nested level and create {@link Schema}
    * object for non-root level.
    */
-  public static void convertFieldToSchemaWithProps(Map<String,JsonNode> fieldProps, Schema targetSchemaObj) {
-    for (Map.Entry<String, JsonNode> stringJsonNodeEntry : fieldProps.entrySet()) {
-      targetSchemaObj.addProp(stringJsonNodeEntry.getKey(), stringJsonNodeEntry.getValue());
+  public static void convertFieldToSchemaWithProps(Map<String,Object> fieldProps,
```

Comment:
Shall we preserve the original method still since it is a public static method ?

Reply:
If we keep two methods one with jsonNode and other with Object, at some point we have to clean this existing method. For example, once we upgrade to Jackson 2.x by default 1.x classes need to be removed. Need to keep the Jackson 1.x classes wherever it is absolutely necessary. For example hive-serde and helix jar has Jackson 1.x dependency. So it would be good to remove this method and it would be helpful the fix the avro changes in a cleaner way and should be able detect issues (if there any) during testing.